scss/_patterns_lists.scss

```diff
@@ -71,14 +71,16 @@ $list-step-bullet-margin: $sph--x-large;
   }
 
   %vf-list-item-ordered {
-    &::marker {
+      counter-reset: li;
```

Comment:
I think the changes you made here are in good direction, but are not necessary in this form.

Overall, I believe the bug comes from the fact that we use 2 different counters in different lists: we use `li` counter and `list-item` counter. The stepped list uses one of them (`li`) and the divided list uses the other (`list-item`).

I believe the difference is not necessary. We can use the same counter in both (so they correctly reset and increment). 

So I think the fix for the issue would be to change any references to counters to use the same one (either `li` or `list-item` - doesn't matter, just the same everywhere).
So any `counter-reset`, `counter-increment` or value of `counter(…)` should use the same (`list-item` for example).